src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/nd4jutils/Nd4jApacheAdapterUtils.java

```diff
@@ -0,0 +1,76 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.nd4jutils;
+
+import org.apache.commons.math3.linear.ArrayRealVector;
+import org.apache.commons.math3.linear.BlockRealMatrix;
+import org.apache.commons.math3.linear.RealMatrix;
+import org.apache.commons.math3.linear.RealVector;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.factory.Nd4j;
+
+import javax.annotation.Nonnull;
+
+/**
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class Nd4jApacheAdapterUtils {
+    /**
+     * INDArray to Apache
+     *
+     * TODO github/gatk-protected issue #853 -- this method must be optimized
```

Comment:
I implemented an optimized version. The idea is to reduce the number of calls to `INDArray` accessors and mutators: to convert to Apache `RealMatrix`, we dump the `INDArray` data buffer to a `double[]` in one shot and use JVM native array copy to quickly reshape it to 2D arrays; then, we readily instantiate a `RealMatrix` out of it. The reverse is more straightforward: we can get the Apache `RealMatrix` data buffer as a `double[][]` and `INDArray` has a constructor for `double[][]`.